### DISCUSSION TITLE
Must add viewer.realize() before EarthManipulator.setViewpoint(Viewpoint)?

### DISCUSSION THREAD

**Discussion by smarttigerfly** [pelicanmapping/osgearth#2248] [Support Q&A]
When I build and run example `osgearth_transform` with `Visual Studio Community 2022` on `Win 10`, the result is incorrect:
```
int
main(int argc, char** argv)
{
    osgEarth::initialize();

    ...
    osg::ref_ptr<osg::Node> model = osgDB::readRefNodeFile("cow.osgt.10000.scale.osgearth_shadergen");
    ...

    osgEarth::Viewpoint vp;
    vp.setNode(app.geo);
    vp.heading()->set(-45.0, Units::DEGREES);
    vp.pitch()->set(-20.0, Units::DEGREES);
    vp.range()->set(model->getBound().radius() * 10.0, Units::METERS);
    em->setViewpoint(vp);

    return viewer.run();
}

```

I run `osgearth_transform.exe` in command-line, the result is:
```
.\osgearth_transform.exe tests/simple.earth
```
![2023-03-26-16-15-43](https://user-images.githubusercontent.com/13800186/227763843-dbef65f6-2277-4c56-8f99-3a871b28dc94.png)

And I add `viewer.realize();` before `em->setViewpoint(vp);`:
```

    ...

    osgEarth::Viewpoint vp;
    vp.setNode(app.geo);
    vp.heading()->set(-45.0, Units::DEGREES);
    vp.pitch()->set(-20.0, Units::DEGREES);
    vp.range()->set(model->getBound().radius() * 10.0, Units::METERS);
    

    viewer.realize();

    em->setViewpoint(vp);

    return viewer.run();

```

the result is:
![2023-03-26-16-10-41](https://user-images.githubusercontent.com/13800186/227763899-d2c3a99f-e1ea-449a-94dc-5ac871ea657e.png)

the statement `em->setViewpoint(vp);`  works.

**Comment by gwaldron** [comment]
I am not able to reproduce this in the main master branch. Which version of osgEarth are you using?

**Reply by smarttigerfly** [reply]
The version is:
```
PS E:\vcpkg> osgearth_version.exe
DisplaySettings::SHADER_GL3
osgEarth 3.3.0 build 135
```
